lib/msf/core/post/dns/resolve_host.rb

```diff
@@ -0,0 +1,71 @@
+# -*- coding: binary -*-
+
+module Msf
+  class Post
+    module DNS
+      ###
+      #
+      # This module resolves session DNS
+      #
+      ###
+      module ResolveHost
+        # Takes the host name and resolves the IP
+        #
+        # @param [String] host
+        # @param [Integer] family 2 for IPV4 and 23 for IPV6
+        # @return [Hash] The resolved IPs
+        def resolve_host(host, family)
+          if client.respond_to?(:net) && client.commands.include?(Rex::Post::Meterpreter::Extensions::Stdapi::COMMAND_ID_STDAPI_NET_RESOLVE_HOST)
+            resolved_host = client.net.resolve.resolve_host(host, family)
+            # We want to only return the array off IPs
+            resolved_host.reject { |k, _v| k == :ip }
+          else
+            ips = []
+            data = cmd_exec("nslookup #{host}")
```

Comment:
`nslookup` is Windows-specific, yet the `resolve_host` method doesn't seem to be platform-specific